Created DSX Data Structure Definition

jaypacamarra opened the pull request:
The DSX data structure is defined as a bit field struct.
In total, we will use 7+6+1+12+4 = 30 bits out of the 32 bits.

I also defined the initializing function which initializes the values of ID, loc, sign, val, and ret to 0. It will take in a pointer to an object of type DSX_data_t.

Commit:
Include *.launch in .gitignore

Commit:
Add data structure in DSX_Library

Commit:
Define DSX bit field data structure

Commit:
Add function to init a dsx data structure by reference

manraj-s commented:
I think the dsx object in main.c should be a volatile variable or volatile pointer

tsaxon commented:
Manraj may be onto something, modifying non-volatile dsx_data in a UART ISR _may_ not update the value in the main superloop? But looks good. 1 question though -> what is "DATA_STRUCTURE_INC_DSX_DATA_STRUCTURE_H_" ?

jaypacamarra commented:
Ya adding volatile will be very good in front of the dsx_data object.

jaypacamarra commented:
DATA_STRUCTURE_INC_DSX_DATA_STRUCTURE_H_

This is a unique identifier for the include guards of that header file. It can be anything. I can see why that would be confusing because that was auto generated. I can change:

DATA_STRUCTURE_INC_DSX_DATA_STRUCTURE_H_     ->    DSX_DATA_STRUCTURE_H

Commit:
Fix dsx_data_structure.c in Inc instead of Src

Commit:
Change header file include guard identifier to be easier to read

Commit:
Add volatile keyword to DSX_data_t

This should be good because the compiler will not optimize the DSX data
structure since it may change in the program. The intended use of the
DSX data structure is constant modification of it from all the command
functions. Adding volatile is super safe thinking.

tsaxon commented:
I see, messed around with the CubeIDE header templates and it gave me a similar long name. Should we shorten those identifiers for any new libraries were making?

jaypacamarra commented:
Yes that would be nice if you shorten the default include guard names. Thanks.

jaypacamarra commented on dsx-fw/Core/Src/main.c:
```diff
@@ -74,7 +74,7 @@ int main(void)
 
 	// Initialize DSX data structure
-	DSX_data_t dsx_data;
+	volatile DSX_data_t dsx_data;
```
Added volatile in front so every functions that take in a DSX_data_t must have volatile in it too.

jaypacamarra commented:
One approval required to merge the pr :)

manraj-s reviewed (approved):
LGTM!
